### ISSUE TITLE
Commit sha isn't listed for unknown

### ISSUE THREAD

**Issue by jsoref** [argoproj/argo-cd#5442]
If you are trying to resolve an environment-specific issue or have a one-off question about the edge case that does not require a feature then please consider asking a question in argocd slack [channel](https://argoproj.github.io/community/join-slack).

Checklist:

* [ ] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [ ] I've included steps to reproduce the bug.
* [ ] I've pasted the output of `argocd version`.

**Describe the bug**

When the system is healthy, it reports the branch and commit sha. When it's in an unknown state, it seems to report the desired branch.

The problem is, I can't tell if it's failing w/ my "fixed" commit, or still failing on the previous commit. I'm assuming that it knows which commit it checked out to try to perform work, even if it didn't like the results.

**To Reproduce**

A list of the steps required to reproduce the issue. Best of all, give us the URL to a repository that exhibits this issue. See #5441

**Expected behavior**

Report the commit sha if applicable, or an error message justifying the lack of availability of such information.

**Screenshots**

![image](https://user-images.githubusercontent.com/2119212/107152638-f422bd80-6936-11eb-9a3e-fc667dd494e5.png)

**Version**

```shell
{
    "Version": "v1.9.0+00f53b5",
    "BuildDate": "2021-01-27T00:04:02Z",
    "GitCommit": "00f53b510d87750492d8446e3303f6ab28258bce",
    "GitTreeState": "clean",
    "GoVersion": "go1.14.12",
    "Compiler": "gc",
    "Platform": "linux/amd64",
    "KsonnetVersion": "v0.13.1",
    "KustomizeVersion": "v3.8.1 2020-07-16T00:58:46Z",
    "HelmVersion": "v3.4.1+gc4e7485",
    "JsonnetVersion": "v0.17.0"
}
```

**Logs**

```
Paste any relevant application logs here.
```

**Comment by jsoref** [comment]
One point of clarity, this was actually a templated application. It was never healthy, but there was definitely a commit sha that was used to try to instantiate the application. It just didn't actually work. It's still important to know which sha that was.

I made a commit which would cause the application not to be created, and I needed to know if that was seen.

**Comment by Eslam-mohammed-anwar** [comment]
Yes we experience the same issue here ,as the argo notification failed to push github statue with error

level=error msg="Failed to notify recipient {github \"\"} defined in resource default/xxxxx-qa: POST https://api.github.com/repos/xxxxx/xxxxx/statuses/%3Cno%20value%3E: 422 No commit found for SHA: <no value> []" resource=default/xxxxxxx-qa
